pcap_manager Compilation error on ubuntu 21.10

**lnav version**
v0.9.0 is the latest

**Describe the bug**
make fails with these errors:
`  CXX      file_collection.o
  CXX      pcap_manager.o
In file included from pcap_manager.cc:39:
pcap_manager.hh:47:10: error: ‘shared_ptr’ in namespace ‘std’ does not name a template type
   47 |     std::shared_ptr<std::vector<std::string>> cr_error_queue;
      |          ^~~~~~~~~~
pcap_manager.hh:41:1: note: ‘std::shared_ptr’ is defined in header ‘<memory>’; did you forget to ‘#include <memory>’?
   40 | #include "auto_fd.hh"
  +++ |+#include <memory>
   41 | //#include <memory>
pcap_manager.cc: In function ‘Result<pcap_manager::convert_result, std::__cxx11::basic_string<char> > pcap_manager::convert(const string&)’:
pcap_manager.cc:126:5: error: too many initializers for ‘pcap_manager::convert_result’
  126 |     });
      |     ^`

**To Reproduce**
Steps to reproduce the behavior:
./autogen.sh
./configure
./make


Workaround:
Add 
`#include <memory> `

to pcap_manager.hh

--- a/docs/source/cli.rst
+++ b/docs/source/cli.rst
@@ -98,6 +98,11 @@ Environment Variables
    If :envvar:`XDG_CONFIG_HOME` is not set, lnav will use this directory
    to store its configuration in a sub-directory named :file:`.lnav`.
 
+.. envvar:: APPDATA
+
+   On Windows, lnav will use this directory instead of HOME
+   to store its configuration in a sub-directory named :file:`.lnav`.
+
 .. envvar:: TZ
 
    The timezone setting is used in some log formats to convert UTC timestamps

--- a/src/base/paths.cc
+++ b/src/base/paths.cc
@@ -29,16 +29,51 @@
 
 #include "config.h"
 
+#ifdef __CYGWIN__
+#include <sstream>
+#include <iostream>
+#endif
+
 #include "paths.hh"
 #include "fmt/format.h"
 
 namespace lnav {
 namespace paths {
 
+#ifdef __CYGWIN__
+char* windows_to_unix_file_path(char* input) {
+    if (input == nullptr) {
+      return nullptr;
+    }
+    std::string file_path;
+    file_path.assign(input);
+
+    // Replace the slashes
+    std::replace(file_path.begin(), file_path.end(), WINDOWS_FILE_PATH_SEPARATOR, UNIX_FILE_PATH_SEPARATOR);
+
+    // Convert the drive letter to lowercase
+    std::transform(file_path.begin(), file_path.begin() + 1, file_path.begin(),
+                   [](unsigned char character) {
+                       return std::tolower(character);
+                   });
+
+    // Remove the colon
+    const auto drive_letter = file_path.substr(0, 1);
+    const auto remaining_path = file_path.substr(2, file_path.size() - 2);
+    file_path = drive_letter + remaining_path;
+
+    std::stringstream stringstream;
+    stringstream << "/cygdrive/";
+    stringstream << file_path;
+
+    return const_cast<char*>(stringstream.str().c_str());
+}
+#endif
+
 ghc::filesystem::path dotlnav()
 {
 #ifdef __CYGWIN__
-    auto home_env = getenv("APPDATA");
+    auto home_env = windows_to_unix_file_path(getenv("APPDATA"));
 #else
     auto home_env = getenv("HOME");
 #endif

--- a/src/base/paths.cc
+++ b/src/base/paths.cc
@@ -37,7 +37,11 @@ namespace paths {
 
 ghc::filesystem::path dotlnav()
 {
+#ifdef __CYGWIN__
+    auto home_env = getenv("APPDATA");
+#else
     auto home_env = getenv("HOME");
+#endif
     auto xdg_config_home = getenv("XDG_CONFIG_HOME");
 
     if (home_env != nullptr) {

--- a/src/base/paths.hh
+++ b/src/base/paths.hh
@@ -39,7 +39,7 @@ namespace paths {
 static const char WINDOWS_FILE_PATH_SEPARATOR = '\\';
 static const char UNIX_FILE_PATH_SEPARATOR = '/';
 
-char* windows_to_unix_file_path(std::string file_path);
+char* windows_to_unix_file_path(char* input);
 #endif
 
 /**

--- a/src/base/paths.hh
+++ b/src/base/paths.hh
@@ -35,6 +35,13 @@
 namespace lnav {
 namespace paths {
 
+#ifdef __CYGWIN__
+static const char WINDOWS_FILE_PATH_SEPARATOR = '\\';
+static const char UNIX_FILE_PATH_SEPARATOR = '/';
+
+char* windows_to_unix_file_path(std::string file_path);
+#endif
+
 /**
  * Compute the path to a file in the user's '.lnav' directory.
  *

--- a/src/lnav.cc
+++ b/src/lnav.cc
@@ -32,6 +32,10 @@
  * a bit.
  */
 
+#ifdef __CYGWIN__
+#include <alloca.h>
+#endif
+
 #include "config.h"
 
 #include <stdio.h>

--- a/src/pcap_manager.cc
+++ b/src/pcap_manager.cc
@@ -33,6 +33,7 @@
 
 #include <vector>
 #include <thread>
+#include <memory>
 
 #include <unistd.h>
 

--- a/src/shlex.cc
+++ b/src/shlex.cc
@@ -29,6 +29,10 @@
  * @file shlex.cc
  */
 
+#ifdef __CYGWIN__
+#include <alloca.h>
+#endif
+
 #include "config.h"
 
 #include "shlex.hh"

--- a/src/string-extension-functions.cc
+++ b/src/string-extension-functions.cc
@@ -7,6 +7,10 @@
  * commercial or non-commercial, and by any means.
  */
 
+#ifdef __CYGWIN__
+#include <alloca.h>
+#endif
+
 #include "config.h"
 
 #include <stdlib.h>

--- a/src/tools/bin2c.c
+++ b/src/tools/bin2c.c
@@ -7,6 +7,10 @@
 // whatever you want with this stuff.  If we meet some day, and you think this stuff is
 // worth it, you can buy me a beer in return.  Sandro Sigala
 
+#ifdef __CYGWIN__
+#include <alloca.h>
+#endif
+
 #include <ctype.h>
 #include <fcntl.h>
 #include <stdio.h>

--- a/src/view_curses.cc
+++ b/src/view_curses.cc
@@ -29,6 +29,10 @@
  * @file view_curses.cc
  */
 
+#ifdef __CYGWIN__
+#include <alloca.h>
+#endif
+
 #include "config.h"
 
 #include <cmath>

--- a/src/yajlpp/json_ptr.cc
+++ b/src/yajlpp/json_ptr.cc
@@ -29,6 +29,10 @@
  * @file json_ptr.cc
  */
 
+#ifdef __CYGWIN__
+#include <alloca.h>
+#endif
+
 #include "config.h"
 
 #include "yajl/api/yajl_gen.h"

--- a/test/drive_data_scanner.cc
+++ b/test/drive_data_scanner.cc
@@ -27,6 +27,10 @@
  * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+#ifdef __CYGWIN__
+#include <alloca.h>
+#endif
+
 #include "config.h"
 
 #include <stdio.h>

